Documentation/concepts/networking/masquerading.rst

```diff
@@ -52,9 +52,90 @@ Implementation Modes
 --------------------
 
 eBPF-based
-  The eBPF-based implementation is the most efficient
-  implementation. It requires Linux kernel 4.19 and can be enabled with the
-  option ``enable-bpf-masquerade: true``.
+**********
+
+The eBPF-based implementation is the most efficient
+implementation. It requires Linux kernel 4.19 and can be enabled with
+the ``global.bpfMasquerade=true`` helm option (enabled by default).
+
+The current implementation depends on :ref:`the BPF NodePort feature <kubeproxy-free>`.
+The dependency will be removed in the Cilium v1.9 release.
+
+Masquerading can take place only on those devices which run the eBPF masquerading
+program. This means that a packet sent from a pod to an outside will be masqueraded
+(to an output device IPv4 address), if the output device runs the program. If not
+specified, the program will be automatically attached to the devices selected by
+:ref:`the BPF NodePort device detection metchanism <Nodeport Devices>`.
+To manually change this, use the ``global.devices`` helm option. ``cilium status``
+can be used to determine on which devices the program is running:
+
+.. parsed-literal::
+
+    kubectl exec -it -n kube-system cilium-xxxxx -- cilium status | grep Masquerading
+    Masquerading:   BPF (ip-masq-agent)   [eth0, eth1]
+
+From the output above, the program is running on the ``eth0`` and ``eth1`` devices.
+
+
+The eBPF-based masquerading can masquerade packets of the following IPv4 L4 protocols:
+
+- TCP
+- UDP
+- ICMP (only Echo request and Echo reply)
+
+By default, any packet from a pod destined to outside of the ``native-routing-cidr``
+range is masqueraded. To allow more fine-grained control, Cilium implements
+`ip-masq-agent <https://github.com/kubernetes-sigs/ip-masq-agent>`_ in eBPF which
+can be enabled with the ``global.ipMasqAgent.enabled=true`` helm option.
+
+The eBPF-based ip-masq-agent supports the ``nonMasqueradeCIDRs`` and
+``masqLinkLocal`` options set in a configuration file. A packet sent from a pod to
+a destination which belongs to any CIDR from the ``nonMasqueradeCIDRs`` is not
+going to be masqueraded. If the configuration file is empty, the agent will provision
+the following non-masquerade CIDRs:
+
+- ``10.0.0.0/8``
+- ``172.16.0.0/12``
+- ``192.168.0.0/16``
+- ``100.64.0.0/10``
+- ``192.0.0.0/24``
+- ``192.0.2.0/24``
+- ``192.88.99.0/24``
+- ``198.18.0.0/15``
+- ``198.51.100.0/24``
+- ``203.0.113.0/24``
+- ``240.0.0.0/4``
+
+In addition, if the ``masqLinkLocal`` is not set or set to false, then
+``169.254.0.0/16`` is appended to the non-masquerade CIDRs list.
+
+The agent uses Fsnotify to track updates to the configuration file, so the original
+``resyncInterval`` option is unnecessary.
+
```

Comment:
```suggestion
```

If it's unnecessary, then why do we document it?

Reply:
Because the vanilla ip-masq-agent requires this option, and I wanted to make clear that we don't poll the config file.